TestModels/SimpleTypes/SimpleBoolean/runtimes/rust/dafny_impl/src/implementation_from_dafny.rs

```diff
@@ -0,0 +1,501 @@
+        -> ::std::rc::Rc<super::r#_Wrappers_Compile::Result<*mut dyn super::r#_simple_dtypes_dsmithyboolean_dinternaldafny_dtypes::ISimpleTypesBooleanClient, ::std::rc::Rc<super::r#_simple_dtypes_dsmithyboolean_dinternaldafny_dtypes::Error>>>{
+            let client: *mut SimpleBooleanClient =
+                ::dafny_runtime::allocate::<SimpleBooleanClient>();
+            SimpleBooleanClient::_ctor(
+                client,
+                &::std::rc::Rc::new(super::r#_SimpleBooleanImpl_Compile::Config::Config {}),
+            );
+            let v = client as *mut dyn super::r#_simple_dtypes_dsmithyboolean_dinternaldafny_dtypes::ISimpleTypesBooleanClient;
+            // build a success
+            ::std::rc::Rc::new(super::r#_Wrappers_Compile::Result::<*mut dyn super::r#_simple_dtypes_dsmithyboolean_dinternaldafny_dtypes::ISimpleTypesBooleanClient, ::std::rc::Rc<super::r#_simple_dtypes_dsmithyboolean_dinternaldafny_dtypes::Error>>::Success{
+          value: v
+      })
+        }
+    }
+
+    struct SimpleBooleanClient {
+        r#_i_config: ::std::rc::Rc<super::r#_SimpleBooleanImpl_Compile::Config>,
+    }
+
+    impl SimpleBooleanClient {
+        fn _ctor(
+            this: *mut SimpleBooleanClient,
+            config: &::std::rc::Rc<super::r#_SimpleBooleanImpl_Compile::Config>,
+        ) {
+            let mut _i_set_config = false;
+            ::dafny_runtime::update_field_uninit!(this, r#_i_config, _i_set_config, config.clone());
+        }
+        fn config(&self) -> ::std::rc::Rc<super::r#_SimpleBooleanImpl_Compile::Config> {
+            self.r#_i_config.clone()
+        }
+    }
+
+    impl super::r#_simple_dtypes_dsmithyboolean_dinternaldafny_dtypes::ISimpleTypesBooleanClient
+        for SimpleBooleanClient
+    {
+        fn GetBoolean(
+            self: &Self,
+            input: &::std::rc::Rc<
+                super::r#_simple_dtypes_dsmithyboolean_dinternaldafny_dtypes::GetBooleanInput,
+            >,
+        ) -> ::std::rc::Rc<
+            super::r#_Wrappers_Compile::Result<
+                ::std::rc::Rc<
+                    super::r#_simple_dtypes_dsmithyboolean_dinternaldafny_dtypes::GetBooleanOutput,
+                >,
+                ::std::rc::Rc<super::r#_simple_dtypes_dsmithyboolean_dinternaldafny_dtypes::Error>,
+            >,
+        > {
+            super::r#_SimpleBooleanImpl_Compile::_default::GetBoolean(&self.config(), input)
+        }
+    }
+    ::dafny_runtime::UpcastTo!(
+        SimpleBooleanClient,
+        dyn super::r#_simple_dtypes_dsmithyboolean_dinternaldafny_dtypes::ISimpleTypesBooleanClient
+    );
+}
+
+mod r#_StandardLibraryInterop_Compile {
+    pub struct WrappersInterop {}
+
+    impl WrappersInterop {
+        pub fn new() -> Self {
+            WrappersInterop {}
+        }
+        pub fn _allocated() -> *mut Self {
+            ::dafny_runtime::allocate::<Self>()
+        }
+        pub fn CreateBooleanSome(
```

Comment:
I totally get why you removed the `CreateStringSome/None` functions, but they are actually coming from here: https://github.com/smithy-lang/smithy-dafny/blob/main-1.x/TestModels/dafny-dependencies/StandardLibrary/src/WrappersInterop.dfy So they will always be emitted by the Dafny compiler not matter what Smithy model we're using, and should still be here.

(I only spotted this by diffing this file against the SimpleString version. :) )

Reply:
I also notice this file has `rustfmt` applied while I think the SimpleString version doesn't, but I think that's actually better since we expect `dafny` to apply `rustfmt` itself anyway, so we should fix the SimpleString version at some point instead.

Reply:
ah, good catch! i've updated the PR to add those back.